src/text/parsers/mod.rs

```diff
@@ -26,8 +26,18 @@ pub(crate) mod timestamp;
 pub(crate) mod top_level;
 pub(crate) mod value;
 
-const WHITESPACE_CHARACTERS: &[char] = &[' ', '\t', '\r', '\n'];
-const WHITESPACE_CHARACTERS_AS_STR: &str = " \t\r\n";
+const WHITESPACE_CHARACTERS: &[char] = &[
+    ' ',        // Space
+    '\t',       // Tab
+    '\r',       // Carriage return
+    '\n',       // Newline
+    '\u{0009}', // Horizontal tab
+    '\u{000B}', // Vertical tab
+    '\u{000C}', // Form feed
```

Comment:
Is there a reason to prefer e.g. `\u{0009}` to `\x09`?

I was curious as the the complete set of built-in escapes like `\n` for Rust since some of these have built-in forms in other languages, so I looked up [ASCII escapes](https://doc.rust-lang.org/beta/reference/tokens.html#ascii-escapes) in the Rust docs.

Reply:
`\x09` would probably be better. I copied these values directly from an `ion-tests` example that was failing.

Incidentally, the reader's handling of Unicode escapes in general isn't quite right. It doesn't handle multi-codepoint escapes properly. When I fix that I'll revisit this.